File: backend/app/schemas/data_management.py
from pydantic import BaseModel


class DataMetric(BaseModel):
    """数据管理页顶部指标。"""

    key: str
    label: str
    value: int | None
    description: str = ""
    status: str = "ready"


class DataSourceItem(BaseModel):
    """数据源概况，统计值来自 PostgreSQL 文档片段表。"""

    name: str
    source_type: str
    storage_targets: list[str]
    document_count: int
    status: str = "ready"


class DataImportStep(BaseModel):
    """数据导入流程步骤。"""

    key: str
    label: str
    description: str
    count: int | None = None
    status: str = "ready"


class DataManagementSummary(BaseModel):
    """数据管理页聚合响应。"""

    metrics: list[DataMetric]
    sources: list[DataSourceItem]
    import_steps: list[DataImportStep]

File: frontend/src/api/dataManagement.ts
import { apiClient } from "./client";
import type { ApiResponse } from "./types";

export interface DataMetric {
  key: string;
  label: string;
  value: number | null;
  description: string;
  status: "ready" | "empty" | "error" | string;
}

export interface DataSourceItem {
  name: string;
  source_type: string;
  storage_targets: string[];
  document_count: number;
  status: string;
}

export interface DataImportStep {
  key: string;
  label: string;
  description: string;
  count: number | null;
  status: string;
}

export interface DataManagementSummary {
  metrics: DataMetric[];
  sources: DataSourceItem[];
  import_steps: DataImportStep[];
}

export async function fetchDataManagementSummary(): Promise<DataManagementSummary> {
  const response = await apiClient.get<ApiResponse<DataManagementSummary>>("/api/data/summary");
  return response.data.data;
}
